spring-beans/src/main/java/org/springframework/beans/factory/annotation/InjectionMetadata.java

```diff
@@ -107,15 +107,20 @@ protected boolean needsRefresh(Class<?> clazz) {
 	}
 
 	public void checkConfigMembers(RootBeanDefinition beanDefinition) {
-		Set<InjectedElement> checkedElements = new LinkedHashSet<>(this.injectedElements.size());
-		for (InjectedElement element : this.injectedElements) {
-			Member member = element.getMember();
-			if (!beanDefinition.isExternallyManagedConfigMember(member)) {
-				beanDefinition.registerExternallyManagedConfigMember(member);
-				checkedElements.add(element);
+		if (this.injectedElements.isEmpty()) {
+			this.checkedElements = Set.of();
+		}
+		else {
+			Set<InjectedElement> checkedElements = new LinkedHashSet<>(this.injectedElements.size()*4/3);
```

Comment:
**size * 4 / 3 + 1** should be better for handling table resize corner cases. (like if the original size is 100, the capacity should be 134 instead of 133).

and for this PR, I don't think we need to specifically handle the empty list case as well as specify the capacity, it makes the code a little bit dirty.

Reply:
>and for this PR, I don't think we need to specifically handle the empty list case as well as specify the capacity, it makes the code a little bit dirty.

The data from memory dump shows otherwise.  Memory is wasted on numerous empty lists